// pogger/dist/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.error = exports.warning = exports.success = exports.info = exports.log = exports.generateDate = exports.errorIcon = exports.warningIcon = exports.successIcon = exports.infoIcon = void 0;
const chalk_1 = require("chalk");
const supports_color_1 = require("supports-color");
exports.infoIcon = supports_color_1.stdout ? "ℹ" : "i";
exports.successIcon = supports_color_1.stdout ? "✔" : "√";
exports.warningIcon = supports_color_1.stdout ? "⚠" : "‼";
exports.errorIcon = supports_color_1.stdout ? "✖" : "×";
exports.generateDate = () => chalk_1.grey(new Date().toLocaleString("en-US", { hour: "numeric", minute: "numeric", second: "numeric", hour12: false }));
exports.log = (logs, icon, color) => console.log(exports.generateDate(), color(icon), ...logs);
exports.info = (...logs) => exports.log(logs, exports.infoIcon, chalk_1.cyan);
exports.success = (...logs) => exports.log(logs, exports.successIcon, chalk_1.green);
exports.warning = (...logs) => exports.log(logs, exports.warningIcon, chalk_1.yellowBright);
exports.error = (...logs) => exports.log(logs, exports.errorIcon, chalk_1.red);
